### DISCUSSION TITLE
Realtime weather data

### DISCUSSION THREAD

**Discussion by mexysfr** [open-meteo/open-meteo#404] [Ideas]
Hi guys !
As always, a huge thanks for letting this repo exists 🙏

Following these points :
- https://github.com/open-meteo/open-meteo/issues/223#issuecomment-1434869806
- https://github.com/open-meteo/open-meteo/issues/255

I was wondering if realtime weather data was planned in a near future ?

I've made a very small list of weather sources (commercial and non-commercial licence), typically list of physical weather stations, that could give some ideas of how to implement this feature in OM.

[Tempest API | Tempest by WeatherFlow](https://business.weatherflow.com/tempest-api)
[Meteociel - Observations météo en temps réel en France - France](https://www.meteociel.fr/observations-meteo/temps-reel.php?region=&duree=1)
[Map view of Holfuy automatic weather stations.](https://holfuy.com/en/map/wind&la=46.00000&lo=3.00000&z=5.3)
[Annuaire des stations amateurs - Infoclimat](https://www.infoclimat.fr/cartes/stations-amateurs.php)
[Netatmo Weathermap](https://weathermap.netatmo.com/)
[Stations en open-data du réseau météorologique Infoclimat (Réseau StatIC) (donnees-stations-infoclimat-static) - data.gouv.fr](https://www.data.gouv.fr/fr/datasets/stations-en-open-data-du-reseau-meteorologique-infoclimat-reseau-static/#/resources)

**Comment by guidocioni** [comment]
The problem with realtime data is that there are so many different sources.
Every country has its own standard on how to define some variables and even for those stations where some international standards exist (e.g. SYNOP) you still have many discrepancies.

I went through this process many time and I think, unfortunately, it is really hard to try and homogenise everything into a single database with common definitions.

**Comment by patrick-zippenfenig** [comment]
At some point in the future I want to implement a measurement database as well. As Guido mentioned it is quite a mess with different standards, formats and licenses. Additionally, I would like to have a distributed/replicatable and scalable database for unstructured timeseries data. 

For the next months, weather models and other gridded datasets will follow. Measurements have a lower priority for me

**Comment by neivnova** [comment]
For measurements there are already some nice open source APIs around. Check e.g. https://meteostat.net/en/

**Comment by ghost** [comment]
I think I mentioned in one of those issues that for the CONUS region there's the RTMA and RTMA-RU models but the data offered by them is quite limited. Looks like in 2025 they're both going to be replaced with the 3DRTMA/3DRTMA-RU model covering North America with similar variables to what HRRR offer. From doing some quick googling it doesn't seem like there's similar models for other regions unfortunately.

**Reply by patrick-zippenfenig** [reply]
In the past days/weeks I integrated 15 minutely `HRRR` data. It is now used for current weather conditions. [See here](https://openmeteo.substack.com/p/sub-hourly-15-minutely-weather-forecasts).

Is `3DRTMA-RU` a dedicated reanalysis/assimilation model that can be downloaded somewhere? It is hard to distinguish between `HRRR, 3DRTMA-RU, HRRRDAS, RTMA and -RU`.

**Reply by ghost** [reply]
The 3DRTMA/3DRTMA-RU models aren't available anywhere yet. I saw in a VLAB discussion somewhere (some reason I cannot find it anymore) that they're hoping to make the hourly 3DRTMA model available in the next few months and I assume the RU version will come closer to release.

RTMA is it's own separate model mainly focusing on current weather conditions and it offers no forecasts. See https://www.emc.ncep.noaa.gov/emc/pages/numerical_forecast_systems/rtma.php for more information.

It's available on AWS here https://registry.opendata.aws/noaa-rtma/ and from NCEP here https://www.nco.ncep.noaa.gov/pmb/products/rtma/, The inventory of data available for the RU version is outdates as since 2020 it does offer cloud clover.
